### ISSUE TITLE
Line number in Foxx Stack Trace does not match

### ISSUE THREAD

**Issue by razvanphp** [arangodb/arangodb#1369]
The exceptions logged with a stack trace contain wrong line numbers. For example, on line 34 of `update-products.js` is a return, not the toString call. I also checked the internal files line number and they seem to be just wrong.

````
2015-06-10T16:02:45 [3132] ERROR Error in foxx route '{ "match" : "/foxxes/scripts/:name", "methods" : [ "post" ], "constraint" : { "name" : "/[^/]+/" } }': 'Cannot read property 'toString' of undefined', Stacktrace: TypeError: Cannot read property 'toString' of undefined
2015-06-10T16:02:45 [3132] ERROR   at anonymous (/home/vagrant/arango-apis/current/foxx/_db/app_dev/test-api/APP/scripts/update-products.js:34:17)
2015-06-10T16:02:45 [3132] ERROR   at [object Object].Module.run (/usr/share/arangodb/js/common/bootstrap/modules.js:1412:10)
2015-06-10T16:02:45 [3132] ERROR   at ArangoApp.loadAppScript (/usr/share/arangodb/js/server/modules/org/arangodb/foxx/arangoApp.js:417:26)
2015-06-10T16:02:45 [3132] ERROR   at executeAppScript (/usr/share/arangodb/js/server/modules/org/arangodb/foxx/manager.js:468:18)
2015-06-10T16:02:45 [3132] ERROR   at Object.runScript (/usr/share/arangodb/js/server/modules/org/arangodb/foxx/manager.js:755:10)
2015-06-10T16:02:45 [3132] ERROR   at /usr/share/arangodb/js/apps/system/_admin/aardvark/APP/foxxes.js:292:28
2015-06-10T16:02:45 [3132] ERROR   at Object.res.action.callback (/usr/share/arangodb/js/server/modules/org/arangodb/foxx/internals.js:101:5)
2015-06-10T16:02:45 [3132] ERROR   at /usr/share/arangodb/js/server/modules/org/arangodb/foxx/routing.js:324:19
````

**Comment by neunhoef** [comment]
Coincidentally, we noticed this yesterday ourselves. The numbers were off by two, if and only if the problem arose from a Foxx app. This is fixed in devel (commit 4214f7882b8bcb9ea151366d75cd53d14f6e4ed8) and will be included in the upcoming 2.6 beta. Thanks for reporting. Do you need a backport in 2.5?

**Comment by pluma** [comment]
@neunhoef I'm fairly certain I introduced the bug in 2.6, so 2.5 should be unaffected.

**Comment by neunhoef** [comment]
Good, even better, no backport needed at all then.

**Comment by razvanphp** [comment]
Ok, I monkey fixed this in my dev machine so I can debug the apps, it seems fixed.

The only strange thing is that I get 3 errors and 3 stack traces for the same error:
````
2015-06-10T16:18:58 [3430] ERROR TypeError: Cannot read property 'toString' of undefined
... stack
2015-06-10T16:18:58 [3430] ERROR Running script 'Update - Products' not possible for mount '/test-api':
2015-06-10T16:18:58 [3430] ERROR [ArangoError 3103: failed to invoke module
... stack
2015-06-10T16:18:58 [3430] ERROR Error in foxx route '{ "match" : "/foxxes/scripts/:name", "methods" : [ "post" ], "constraint" : { "name" : "/[^/]+/" } }': 'Cannot read property 'toString' of undefined', Stacktrace: TypeError: Cannot read property 'toString' of undefined
... stack
````

is this intended? It might be a little confusing..

**Comment by fceller** [comment]
Yes, because the code in executed in different V8 contexts. This causes it to be displayed multiple times.

**Comment by pluma** [comment]
@fceller I think @razvanphp is talking about getting multiple stacktraces for the same error. I've cleaned that part up a bit last week because the noise from log-and-rethrow did tend to get a bit excessive. It should be fixed in the next 2.6 release.
